core/src/main/scala/chisel3/ModuleImpl.scala

```diff
@@ -908,6 +931,18 @@ package experimental {
           case None    => getRef.name
           case Some(c) => getRef.fullName(c)
         }
+  }
+}
 
+object withModulePrefix {
+  def apply[T](prefix: String)(block: => T): T = {
+    Builder.pushModulePrefix(prefix)
+    val res = block // execute block
+    Builder.popModulePrefix()
+    res
   }
 }
+
+case class ModulePrefixAnnotation(target: IsModule, prefix: String) extends SingleTargetAnnotation[IsModule] {
+  def duplicate(n: IsModule): ModulePrefixAnnotation = this.copy(target = n)
+}
```

Comment:
Can this be made private? There's no need to make this publicly accessible. It also lets us delete it in the future.